core/templates/dev/head/pages/exploration_editor/simple_editor_tab/SimpleEditorManagerService.js

```diff
@@ -138,18 +145,54 @@ oppia.factory('SimpleEditorManagerService', [
           data.questionList.isEmpty() ?
           SimpleEditorShimService.getInitStateName() :
           data.questionList.getLastQuestion().getDestinationStateName());
-
         SimpleEditorShimService.saveInteractionId(
-          lastStateName, DEFAULT_INTERACTION.ID);
+          lastStateName, 'MultipleChoiceInput');
         SimpleEditorShimService.saveCustomizationArgs(
-          lastStateName, DEFAULT_INTERACTION.CUSTOMIZATION_ARGS);
+          lastStateName,
+          DEFAULT_INTERACTION_PROPERTIES.MultipleChoiceInput.CUSTOMIZATION_ARGS
+        );
         SimpleEditorShimService.saveDefaultOutcome(
           lastStateName, OutcomeObjectFactory.createEmpty(lastStateName));
 
         var stateData = SimpleEditorShimService.getState(lastStateName);
         data.questionList.addQuestion(QuestionObjectFactory.create(
           lastStateName, stateData.interaction, ''));
       },
+      changeQuestion: function(newQuestionType, index) {
+        var currentStateName = data.questionList.getAllStateNames()[index];
+        var nextStateName = data.questionList.getAllStateNames()[index + 1];
+        var questionCount = data.questionList.getQuestionCount();
+        var currentInteractionId = (
+          SimpleEditorShimService.getInteractionId(currentStateName));
+        var doesLastQuestionHaveAnswerGroups = (
+          QuestionListObjectFactory.doesLastQuestionHaveAnswerGroups);
+
+        // Update Question Type If interactionId is not same.
+        if (newQuestionType !== currentInteractionId) {
+          var newAnswerGroups = [];
+          SimpleEditorShimService.saveInteractionId(
+            currentStateName, newQuestionType);
+          SimpleEditorShimService.saveCustomizationArgs(
+            currentStateName,
+            DEFAULT_INTERACTION_PROPERTIES[newQuestionType].CUSTOMIZATION_ARGS);
+          newAnswerGroups.push(AnswerGroupObjectFactory.createNew([
+            RuleObjectFactory.createNew(
+              DEFAULT_INTERACTION_PROPERTIES[newQuestionType].
+                FIRST_ANSWER_GROUP_RULE.type,
+              DEFAULT_INTERACTION_PROPERTIES[newQuestionType].
+                FIRST_ANSWER_GROUP_RULE.value
+            )
+          ], OutcomeObjectFactory.createEmpty(nextStateName), false));
+
+          if (doesLastQuestionHaveAnswerGroups && index !== questionCount - 1) {
```

Comment:
Ok let's assume you have clicked on the add question and haven't selected any answer. At this point, next state is not getting created. If we're going to save answer group pointing to next state we're getting dest validation error, so we're not saving answer group in when it's the last question having no answer group.